core/src/main/scala/dagr/core/execsystem/TaskManager.scala

```diff
@@ -635,20 +511,18 @@ class TaskManager(taskManagerResources: TaskManagerResources = TaskManagerDefaul
     logger.debug("runSchedulerOnce: finishing one round of execution")
     logger.debug("runSchedulerOnce: found " + runningTasks.size + " running tasks and " + tasksToSchedule.size + " tasks to schedule")
 
-    (readyTasks, tasksToSchedule.keys.toList, runningTasks.keys.toList, completedTasks.keys.map(taskId => taskValueFor(taskId)).toList)
+    (
+      readyTasks,
+      tasksToSchedule.keys.toList,
```

Comment:
Just wondering if the API for this method might be loosened to returning `Iterable` or `Traversable` for some of these, since it seems like the ordering is not important and then you could do this without generating temporary lists every iteration.